### DISCUSSION TITLE
Direnv support?

### DISCUSSION THREAD

**Discussion by johnhamelink** [casey/just#1450] [Q&A]
Hi there,

I was wondering if it's possible for Just's dot-env integration could be used to provide direnv (`.envrc`) support? `.envrc` files are just shell scripts.

https://direnv.net/

**Comment by casey** [comment]
I don't think Just would directly support direnv, but it sounds like it might be supportable with a non-direnv-specific feature. What kind of integration were you thinking of?

**Reply by johnhamelink** [reply]
@casey I’m looking for the same functionality as dotenv has, ideally: you reference direnv in the justfile somehow, the envrc is evaluated, the resultant env vars are passed on to relevant recipes.

**Reply by casey** [reply]
Don't direnv files automatically get executed when you're in the same directory, thus making the variables available in a just invocation in the same directory?

**Reply by johnhamelink** [reply]
@casey you are correct - I think on reflection this isn't a huge issue for my usecase. Perhaps others might see differently.

**Reply by casey** [reply]
I just mean that if you already use direnv, won't it work with just without special support, since just will inherit the environment that direnv creates?

**Reply by johnhamelink** [reply]
@casey yes, but I’m running just from Emacs with Justl, so by default my shell is not evaluating things. I can overcome this by using https://github.com/purcell/envrc to seed the environment
